File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/util/CollectionUtils.java
package io.github.potjerodekool.openapi.internal.util;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class CollectionUtils {

    private CollectionUtils() {
    }

    public static <T> List<T> nonNull(final List<T> list) {
        return list != null
                ? list
                : new ArrayList<>();
    }

    public static <K,V> Map<K,V> nonNull(final Map<K, V> map) {
        return map != null
                ? map
                : new HashMap<>();
    }
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/util/CollectionUtils.kt
package io.github.potjerodekool.openapi.common.util

object CollectionUtils {
    @JvmStatic
    fun <T> nonNull(list: List<T>?): List<T> {
        return list ?: ArrayList()
    }

    @JvmStatic
    fun <K, V> nonNull(map: Map<K, V>?): Map<K, V> {
        return map ?: HashMap()
    }
}
